motion_planning/include/planner_service.hpp

```diff
@@ -0,0 +1,37 @@
+#include <ml4kp_bridge/defs.h>
+
+#include <ros/ros.h>
+
+namespace mj_ros
+{
+template <typename PlannerPtr, typename Service>
+class planner_service_t
+{
+private:
+  ros::ServiceServer _service_server;
+  PlannerPtr _planner;
+  Service _service;
+
+public:
+  planner_service_t(ros::NodeHandle& nh, PlannerPtr planner) : _planner(planner)
+  {
+    const std::string root{ ros::this_node::getNamespace() };
+    const std::string service_name{ root + "/planner_service" };
+    _service_server = nh.advertiseService(service_name, &planner_service_t::service_callback, this);
+  }
+
+  void run()
+  {
+    ros::spin();
```

Comment:
This shouldn't be needed as long as there is a call to `ros::AsyncSpinner spinner(1);` in the `main()`